### ISSUE TITLE
Crash when response is multiple packets

### ISSUE THREAD

**Issue by tannewt** [adafruit/Glider-for-iOS#1]
The file transfer protocol commonly has header responses with a variable portion after them. The variable portion may or may not be part of the packet received by the app. Internally in CP I'm just appending incoming packets until the full variable portion has been received.

The iOS code will crash in this case now because it tries to slice past the end of the data with `subdata`. For example, read does this at line 391 in BlePeripheral+FileTransfer.swift.

**Comment by antonio-openroad** [comment]
Thanks for the info.
I supposed that a packet would always contain a complete header+variable portion.
I will change how the packets are processed to take this into account. 

Do you have an example to reproduce the error so I can test that it is fixed once I implement it?

**Comment by tannewt** [comment]
I think any read with a chunk size greater than 512 will trigger this. Packets will be <512 due to characteristic limits.

**Comment by antonio-openroad** [comment]
I tried setting a read chunk size of 513, but  I get an error on the Python side and I dont receive the response. 
The error is: ValueError: Total data to write is larger than outgoing_packet_length
(which actually makes sense)

Anyway, I changed the code to process the chunks only if I recevied enouth data or accumulate it until I have enought data. 
Also added some checks to avoid race conditions, like receiving code while processing the previous received chunk. 

I think it should work, but I'm not 100% sure. 
I will add it as a PR, and we can do more tests
